agent/config/config.go

```diff
@@ -609,10 +609,11 @@ type AutoEncrypt struct {
 type Connect struct {
 	// Enabled opts the agent into connect. It should be set on all clients and
 	// servers in a cluster for correct connect operation.
-	Enabled                         *bool                  `mapstructure:"enabled" json:"enabled,omitempty"`
-	CAProvider                      *string                `mapstructure:"ca_provider" json:"ca_provider,omitempty"`
-	CAConfig                        map[string]interface{} `mapstructure:"ca_config" json:"ca_config,omitempty"`
-	MeshGatewayWANFederationEnabled *bool                  `mapstructure:"enable_mesh_gateway_wan_federation" json:"enable_mesh_gateway_wan_federation,omitempty"`
+	Enabled                                 *bool                  `mapstructure:"enabled" json:"enabled,omitempty"`
+	CAProvider                              *string                `mapstructure:"ca_provider" json:"ca_provider,omitempty"`
+	CAConfig                                map[string]interface{} `mapstructure:"ca_config" json:"ca_config,omitempty"`
+	MeshGatewayWANFederationEnabled         *bool                  `mapstructure:"enable_mesh_gateway_wan_federation" json:"enable_mesh_gateway_wan_federation,omitempty"`
+	DefaultDeregisterCriticalServiceAfter   *string                `mapstructure:"default_deregister_critical_service_after,omitempty"`
```

Comment:
No error in the field type?

Reply:
> No error in the field type?

No because we require a string, but afterward load it using durationVal() function, which does the type cast in the appropriate type to store in the Runtime object.